File: lab06/all_whales.pl
#!/usr/bin/perl -w 
%result = ();
while ($input = <STDIN>) {
    $input = lc $input;
    @number = split / /, $input;
    $input =~ s/^\d+//g;
    $input =~ s/\s+/ /g;
    $input =~ s/^\s//g;
    $input =~ s/\s$//g;
    $input =~ s/s$//g;
    if (exists $result{$input}) {
	$result{$input}[0] += $number[0];
	$result{$input}[1] += 1;
    }
    else {
	$result{$input}[0] = $number[0];
	$result{$input}[1] = 1;
    }
}
foreach $key (sort keys %result) 
{
    $s = $key;
    chomp $s;
    print "$s observations: $result{$key}[1] pods, $result{$key}[0] individuals\n";
}      

File: lab06/all_whales.py
#!/usr/bin/python3
import sys
import re
result = {}
name=[]
try:
    while 1:
        x = str(input())
        x = x.lower()
        number = x.split(' ')
        x = re.sub('^\d+|\s+|^\s+|\s$|s$','',x)
        if x in result:
            result[x][0] += int(number[0])
            result[x][1] += 1
        else:
            result[x] = []
            result[x].append(int(number[0]))
            result[x].append(1)
            name.append(x)
except EOFError:
    name = sorted(name)
    for x in name:
        print('{} observations: {} pods, {} individuals'.format(x,result[x][1],result[x][0]))
    exit()
        
